test/Amazon.AspNetCore.Identity.Cognito.Test/CognitoUserManagerTest.cs

```diff
@@ -0,0 +1,295 @@
+        {
+            userStoreMock.Setup(mock => mock.SetPhoneNumberAsync(It.IsAny<CognitoUser>(), It.IsAny<string>(), It.IsAny<CancellationToken>())).Returns(Task.FromResult(0)).Verifiable();
+            userStoreMock.Setup(mock => mock.UpdateAsync(It.IsAny<CognitoUser>(), It.IsAny<CancellationToken>())).Returns(Task.FromResult(IdentityResult.Success)).Verifiable();
+            var output = await userManager.SetPhoneNumberAsync(GetCognitoUser(), "+1234567890").ConfigureAwait(false);
+            Assert.Equal(IdentityResult.Success, output);
+            userStoreMock.Verify();
+        }
+
+        [Fact]
+        public async void Test_GivenAnUser_WhenSetEmail_ThenResponseIsNotAltered()
+        {
+            userStoreMock.Setup(mock => mock.SetEmailAsync(It.IsAny<CognitoUser>(), It.IsAny<string>(), It.IsAny<CancellationToken>())).Returns(Task.FromResult(0)).Verifiable();
+            userStoreMock.Setup(mock => mock.UpdateAsync(It.IsAny<CognitoUser>(), It.IsAny<CancellationToken>())).Returns(Task.FromResult(IdentityResult.Success)).Verifiable();
+            var output = await userManager.SetEmailAsync(GetCognitoUser(), "darth.vader@amazon.com").ConfigureAwait(false);
+            Assert.Equal(IdentityResult.Success, output);
+            userStoreMock.Verify();
+        }
+
+        [Fact]
+        public async void Test_GivenAnUser_WhenUpdateUser_ThenResponseIsNotAltered()
+        {
+            userStoreMock.Setup(mock => mock.UpdateAsync(It.IsAny<CognitoUser>(), It.IsAny<CancellationToken>())).Returns(Task.FromResult(IdentityResult.Success)).Verifiable();
+            var output = await userManager.UpdateAsync(GetCognitoUser()).ConfigureAwait(false);
+            Assert.Equal(IdentityResult.Success, output);
+            userStoreMock.Verify();
+        }
+
+        [Fact]
+        public async void Test_GivenAnUser_WhenAddClaims_ThenResponseIsNotAltered()
+        {
+            userStoreMock.Setup(mock => mock.AddClaimsAsync(It.IsAny<CognitoUser>(), It.IsAny<IEnumerable<Claim>>(), It.IsAny<CancellationToken>())).Returns(Task.FromResult(IdentityResult.Success)).Verifiable();
+            var output = await userManager.AddClaimsAsync(GetCognitoUser(), new List<Claim>()).ConfigureAwait(false);
+            Assert.Equal(IdentityResult.Success, output);
+            userStoreMock.Verify();
+        }
+
+        [Fact]
+        public async void Test_GivenAnUser_WhenRemoveClaims_ThenResponseIsNotAltered()
+        {
+            userStoreMock.Setup(mock => mock.RemoveClaimsAsync(It.IsAny<CognitoUser>(), It.IsAny<IEnumerable<Claim>>(), It.IsAny<CancellationToken>())).Returns(Task.FromResult(IdentityResult.Success)).Verifiable();
+            var output = await userManager.RemoveClaimsAsync(GetCognitoUser(), new List<Claim>()).ConfigureAwait(false);
+            Assert.Equal(IdentityResult.Success, output);
+            userStoreMock.Verify();
+        }
+
+        #region ExceptionTests
+
+        [Fact]
```

Comment:
is it really worth making these individual tests?  Why not make it into a single or a few grouped tests?

Reply:
I think it's cleaner to split the tests, this way the mocking behavior is more readable.